api/capi/src/nntrainer.cpp

```diff
@@ -99,6 +99,30 @@ template <typename F> static int nntrainer_exception_boundary(F &&func) {
   }
 }
 
+/**
+ * @brief std::make_shared wrapped with exception boundary
+ *
+ * @tparam Tv value type.
+ * @tparam Tp pointer type.
+ * @tparam Types args used to construct
+ * @param target pointer
+ * @param args args
+ * @return int error value. ML_ERROR_OUT_OF_MEMORY if fail
+ */
+template <typename Tv, typename Tp, typename... Types>
+static int exception_bounded_make_shared(Tp &target, Types... args) {
+  int status = ML_ERROR_NONE;
+
+  try {
+    target = std::make_shared<Tv>(args...);
+  } catch (std::bad_alloc &e) {
```

Comment:
as this calls the constuctor, there might be other exceptions thrown as well.
they remain uncaught with this.

Reply:
Updated tto use nntrainer_exception_boundary